internal/resources/storage.go

```diff
@@ -73,7 +74,7 @@ func (b *StorageClusterBuilder) GetGRPCEndpointWithProto() string {
 	return fmt.Sprintf("%s%s", proto, b.GetGRPCEndpoint())
 }
 
-func (b *StorageClusterBuilder) GetResourceBuilders() []ResourceBuilder {
+func (b *StorageClusterBuilder) GetResourceBuilders(restConfig *rest.Config) []ResourceBuilder {
```

Comment:
I had to do a minor dirty propagation of `restConfig` entity for querying k8s api about secrets on the stage of building resources. It changed the signature of quite a few functions. If you have any ideas on how to do it a bit cleaner, that would be great!